src/Examine.Lucene/Indexing/DateTimeType.cs

```diff
@@ -11,15 +12,18 @@ public class DateTimeType : IndexFieldRangeValueType<DateTime>
     {
         public DateResolution Resolution { get; }
 
+        private readonly bool _isFacetable;
+
         /// <summary>
         /// Can be sorted by the normal field name
         /// </summary>
         public override string SortableFieldName => FieldName;
 
-        public DateTimeType(string fieldName, ILoggerFactory logger, DateResolution resolution, bool store = true)
+        public DateTimeType(string fieldName, ILoggerFactory logger, DateResolution resolution, bool store = true, bool isFacetable = false)
```

Comment:
Unfortunately we'll actually have to create overloads of the ctor's here instead of having an optional parameter. This is for backwards compatibility. Adding an optional parameter actually breaks the binary contract of this constructor, so anyone using the previous constructor (everyone) would break if their project wasn't rebuilt.

Reply:
Should the same be done for the searchers constructors? Maybe by making a new constructor accepting facetsconfig and marking the old one as obsolete? This would also avoid the current breaking change.

And what are your thoughts on Callums comment? Personally I think the suggestions are great and the pr could be refactored with the suggestions.

(Although the features suggestions I'm not sure Lucene Facets support. At least I don't know how to implement them 😅) CC: @callumbwhyte

Reply:
@Shazwazza Then the ctor's would look like this right?

```csharp
public DateTimeType(string fieldName, ILoggerFactory logger, DateResolution resolution, bool store = true)
    : base(fieldName, logger, store)
{
    Resolution = resolution;
    _isFacetable = false;
}

public DateTimeType(string fieldName, ILoggerFactory logger, DateResolution resolution, bool store = true, bool isFacetable = false)
    : base(fieldName, logger, store)
{
    Resolution = resolution;
    _isFacetable = isFacetable;
}
```

Wouldn't this make the first ctor obsolete and should it be marked like this? I'm also wondering if it would even be possible to use a new version of Examine without rebuilding your project. Isn't it normal to always rebuild a project when a package is upgraded?

Reply:
Due to the way that optional params work, the new overload would have to be:

public DateTimeType(string fieldName, ILoggerFactory logger, DateResolution resolution, bool store, bool isFacetable)

I'll update them all.